clientd3d/list.c

```diff
@@ -237,6 +237,25 @@ void *list_find_item(list_type l, void *data, CompareProc compare)
    return NULL;
 }
 /************************************************************************/
+/*
+ * list_get_position: Find and return the position of an item in the given list.
+ *   The compare function is used to determine if two items are equal.
+ *   Returns NULL if item not found.
```

Comment:
Yikes, this is super broken.  NULL is the same as 0, so this function actually returns 0 for any item not in the list.

I suggest returning -1 instead, and checking for that value everywhere this is called.

Reply:
It looks like there are a number of functions within this clientd3d/list.c file that return NULL

- list_first_item
- list_last_item
- list_delete_item
- list_delete
- list_destroy

Should I

1. make the requested change of returning -1 for the new list_get_position function only?
2. return -1 for list_get_position but also fix the other functions?
3. or not make this change so it returns NULL just as the other functions do?

Reply:
Other functions return NULL because it makes sense as a void * or list_type value.  They are not broken. 

However, this function returns an index, which is an int.  Since NULL is the same as 0, your #3 proposes returning 0, meaning, "item is in the first position in the list", when the item is not in the list.  That is surely not the behavior you want.  So I'd return -1 instead.

Reply:
Understood.  This is now fixed.  I have the function return -1 now, and references check for a -1 value.